### ISSUE TITLE
Why does  LSTM model not have  sigmoid fuction?

### ISSUE THREAD

**Issue by wangyanhui666** [cvqluu/nn-similarity-diarization#4]
class LSTMSimilarity(nn.Module):

    def __init__(self, input_size=256, hidden_size=256, num_layers=2):
        super(LSTMSimilarity, self).__init__()
        self.lstm = nn.LSTM(input_size,
                            hidden_size,
                            num_layers=num_layers,
                            bidirectional=True,
                            batch_first=True)
        self.fc1 = nn.Linear(hidden_size*2, 64)
        self.nl = nn.ReLU(inplace=True)
        self.fc2 = nn.Linear(64, 1)

    def forward(self, x):
        self.lstm.flatten_parameters()
        x, _ = self.lstm(x)
        x = self.fc1(x)
        x = self.nl(x)
        x = self.fc2(x).squeeze(2)
        return x

this is your LSTM model, but in original paper, it need a sigmoid fuction before output, like this:

x = torch.sigmoid(self.fc2(x).squeeze(2))
        return x
but when I change this, totally error change from 9.7% to 37%, I don't know why

**Comment by cvqluu** [comment]
I assumed you worked this out if the issue is now closed, but using `nn.BCEWithLogitsLoss` as the criterion in `train.py` combines the log loss and sigmoid in one function.

You can look here for more details: https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html

**Comment by wangyanhui666** [comment]
Yes, I already know. Thank you vary much!
